functions/webhooks/serviceConversationListener.protected.ts

```diff
@@ -0,0 +1,125 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import { Context, ServerlessCallback } from '@twilio-labs/serverless-runtime-types/types';
+import { responseWithCors, bindResolve, error500 } from '@tech-matters/serverless-helpers';
+
+export type ConversationSid = `CH${string}`;
+export type ParticipantSid = `MB${string}`;
+
+type ServiceConversationListenerEvent = {
+  Body: string;
+  ParticipantSid: ParticipantSid;
+  ConversationSid: ConversationSid;
+  EventType: string;
+  MessageSid: string;
+};
+
+export type Body = ServiceConversationListenerEvent;
+
+export const sendConversationMessage = async (
+  context: Context,
+  {
+    conversationSid,
+    author,
+    messageText,
+    messageAttributes,
+  }: {
+    conversationSid: ConversationSid;
+    author: string;
+    messageText: string;
+    messageAttributes?: string;
+  },
+) =>
+  context
+    .getTwilioClient()
+    .conversations.conversations.get(conversationSid)
+    .messages.create({
+      body: messageText,
+      author,
+      xTwilioWebhookEnabled: 'true',
+      ...(messageAttributes && { attributes: messageAttributes }),
+    });
+
+const getTimeDifference = async (isoString: Date): Promise<string> => {
+  const unitCheck = (count: number, unit: string) => (count > 1 ? `${unit}s` : unit);
+  // Create a new Date object from the ISO string
+  const givenDate = new Date(isoString);
+
+  // Get the current date and time
+  const currentDate = new Date();
+
+  // Calculate the difference in milliseconds
+  const differenceInMillis = currentDate.getTime() - givenDate.getTime();
+
+  // Convert the difference to seconds
+  const differenceInSeconds = Math.floor(differenceInMillis / 1000);
+  const differenceInMinutes = Math.floor(differenceInSeconds / 60);
+
+  // Determine whether to return the difference in seconds or minutes
+  return differenceInSeconds < 60
+    ? `${differenceInSeconds} ${unitCheck(differenceInSeconds, 'second')}`
+    : `${differenceInMinutes} ${unitCheck(differenceInMinutes, 'minute')}`;
+};
+
+export const handler = async (context: Context, event: Body, callback: ServerlessCallback) => {
+  const response = responseWithCors();
+  const resolve = bindResolve(callback)(response);
+  try {
+    const { EventType, ConversationSid, MessageSid, ParticipantSid, Body } = event;
+
+    let messageAuthor: string | undefined;
+
+    // check if it's the onMessageAdded event
+    if (EventType === 'onMessageAdded') {
+      const conversationMessage = await context
+        .getTwilioClient()
+        .conversations.v1.conversations(ConversationSid)
+        .messages(MessageSid)
+        .fetch();
+
+      const participantsList = await context
+        .getTwilioClient()
+        .conversations.v1.conversations(ConversationSid)
+        .participants.list();
+
+      participantsList.forEach((participant) => {
```

Comment:
In the custom channel logic, we check the `participantSid` attribute of the conversation (under the `attributes` property, not the participant list). It would be safer to replicate the logic we know is working there than doing something new here